File: common/src/main/java/net/impleri/playerskills/integration/kubejs/basic/BasicSkillJS.java
package net.impleri.playerskills.integration.kubejs.basic;

import net.impleri.playerskills.basic.BasicSkill;
import net.impleri.playerskills.integration.kubejs.skills.GenericSkillBuilderJS;
import net.minecraft.resources.ResourceLocation;

public class BasicSkillJS extends BasicSkill {
    public BasicSkillJS(Builder builder) {
        super(builder.id, builder.initialValue, builder.description, builder.options, builder.changesAllowed);
    }

    public static class Builder extends GenericSkillBuilderJS<Boolean> {
        public Builder(ResourceLocation name) {
            super(name);
        }

        @Override
        public BasicSkillJS createObject() {
            return new BasicSkillJS(this);
        }
    }
}

File: common/src/main/kotlin/net/impleri/playerskills/integrations/kubejs/skills/BasicSkillJS.kt
package net.impleri.playerskills.integrations.kubejs.skills

import net.impleri.playerskills.api.Skill
import net.impleri.playerskills.skills.basic.BasicSkill
import net.minecraft.resources.ResourceLocation

class BasicSkillJS(builder: Builder) : BasicSkill(
  builder.id,
  builder.initialValue,
  builder.description,
  builder.options,
  builder.changesAllowed,
  builder.teamMode,
  builder.notify,
  builder.notifyKey
) {
  class Builder(name: ResourceLocation) : GenericSkillBuilderJS<Boolean>(name) {
    override fun createObject(): Skill<Boolean> {
      return BasicSkillJS(this)
    }
  }
}
